src/widgets/RRule.tsx

```diff
@@ -0,0 +1,249 @@
+        }}
+      >
+        {menuItemsFrequency}
+      </Select>
+      <Select
+        value={getEnds()}
+        onChange={(event: React.FormEvent<{ value: unknown }>) => {
+          const value = (event.target as HTMLSelectElement).value;
+          switch (value) {
+            case 'onDate':
+              updateRule({ count: null, until: new Date() });
+              break;
+            case 'after':
+              updateRule({ until: undefined, count: 1 });
+              break;
+            default:
+              updateRule({ count: null, until: undefined });
+              break;
+          }
+        }}>
+        <MenuItem value="never">Never</MenuItem>
+        <MenuItem value="after">After</MenuItem>
+        <MenuItem value="onDate">On Date</MenuItem>
+      </Select>
+
+      <TextField
+        type="number"
+        placeholder="Interval"
+        inputProps={{ min: 1, max: 1000 }}
+        value={options.interval}
+        onChange={(event: React.FormEvent<{ value: unknown }>) => {
+          event.preventDefault();
+          const inputNode = event.currentTarget as HTMLInputElement;
+          if (inputNode.value === '') {
+            updateRule({ interval: undefined });
+          } else if (inputNode.valueAsNumber) {
+            updateRule({ interval: inputNode.valueAsNumber });
+          }
+        }}
+      />
+
+      {(options.freq === frequency['weekly']) &&
+        <FormGroup
+          row>
+          {checkboxWeekDays}
+        </FormGroup>
+      }
+      {(options.freq === frequency['monthly'] || options.freq === frequency['yearly']) &&
+        <Select
+          value={getMonthReapetType()}
+          onChange={(event: React.FormEvent<{ value: unknown }>) => {
+            const value = (event.target as HTMLInputElement).value;
+            if (value === 'bymonthday') {
+              updateRule({ bymonthday: 1, bysetpos: null, bymonth: null });
+            } else if (value === 'bysetpos') {
+              updateRule({ bymonthday: null, bysetpos: 1 });
+            }
+          }}
+        >
+          <MenuItem value="bymonthday">On day</MenuItem>
+          <MenuItem value="bysetpos">On the</MenuItem>
+        </Select>
+      }
+      {options.bymonthday &&
+        <Select
+          value={options.bymonthday || 1}
+          onChange={(event: React.FormEvent<{ value: unknown }>) => {
+            updateRule({ bymonthday: Number((event.target as HTMLInputElement).value) });
+          }}
+        >
+          {menuItemDaysNumbers}
+        </Select>
+      }
+      {options.bysetpos &&
+        <Select
+          value={options.bysetpos}
+          onChange={(event: React.FormEvent<{ value: unknown }>) => {
+            updateRule({ bysetpos: Number((event.target as HTMLInputElement).value) });
+          }}>
+          <MenuItem value={1}>First</MenuItem>
+          <MenuItem value={2}>Second</MenuItem>
+          <MenuItem value={3}>Third</MenuItem>
+          <MenuItem value={4}>Foruth</MenuItem>
```

Comment:
Type, should be Fourth I assume?

Reply:
Should it be a set of values like this though? Not an arbitrary number input? I'm asking, I have no idea.

Reply:
Leave it a selection like this for now actually, don't worry about arbitrary number inputs.